tests/network/bindingplugin_macvtap.go

```diff
@@ -0,0 +1,251 @@
+				macAddress := macAddressHW.String()
+
+				serverVMI, err = createFedoraVMIRandomNode(macvtapNetworkName, macAddress)
+				Expect(err).NotTo(HaveOccurred(), "must have succeeded creating a fedora VMI on a random node")
+				Expect(serverVMI.Status.Interfaces).NotTo(BeEmpty(), "a migrate-able VMI must have network interfaces")
+				serverVMIPodName = tests.GetVmPodName(kubevirt.Client(), serverVMI)
+
+				serverIP, err = waitVMMacvtapIfaceIPReport(serverVMI, macAddress, macvtapIfaceIPReportTimeout)
+				Expect(err).NotTo(HaveOccurred(), "should have managed to figure out the IP of the server VMI")
+			})
+
+			BeforeEach(func() {
+				// TODO test also the IPv6 address (issue- https://github.com/kubevirt/kubevirt/issues/7506)
+				libnet.SkipWhenClusterNotSupportIpv4()
+				Expect(libnet.PingFromVMConsole(clientVMI, serverIP)).To(Succeed(), "connectivity is expected *before* migrating the VMI")
+			})
+
+			It("should keep connectivity after a migration", func() {
+				const containerCompletionWaitTime = 60
+				migration := libmigration.New(serverVMI.Name, serverVMI.GetNamespace())
+				_ = libmigration.RunMigrationAndExpectToCompleteWithDefaultTimeout(kubevirt.Client(), migration)
+				// In case of clientVMI and serverVMI running on the same node before migration, the serverVMI
+				// will be reachable only when the original launcher pod terminates.
+				Eventually(func() error {
+					return waitForPodCompleted(serverVMI.Namespace, serverVMIPodName)
+				}, containerCompletionWaitTime, time.Second).Should(Succeed(), fmt.Sprintf("all containers should complete in source virt-launcher pod: %s", serverVMIPodName))
+				Expect(libnet.PingFromVMConsole(clientVMI, serverIP)).To(Succeed(), "connectivity is expected *after* migrating the VMI")
+			})
+		})
+	})
+})
+
+func createAlpineVMIStaticIPOnNode(nodeName string, networkName string, ifaceName string, ipCIDR string, mac *string) *v1.VirtualMachineInstance {
+	var vmi *v1.VirtualMachineInstance
+	if mac != nil {
+		vmi = libvmi.NewAlpineWithTestTooling(
+			libvmi.WithInterface(*libvmi.InterfaceWithMac(libvmi.InterfaceWithMacvtapBindingPlugin(networkName), *mac)),
+			libvmi.WithNetwork(libvmi.MultusNetwork(networkName, networkName)),
+			libvmi.WithNodeAffinityFor(nodeName),
+		)
+	} else {
+		vmi = libvmi.NewAlpine(
+			libvmi.WithInterface(*libvmi.InterfaceWithMacvtapBindingPlugin(networkName)),
+			libvmi.WithNetwork(libvmi.MultusNetwork(networkName, networkName)),
+			libvmi.WithNodeAffinityFor(nodeName),
+		)
+	}
+	vmi, err := kubevirt.Client().VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi)
+	ExpectWithOffset(1, err).ToNot(HaveOccurred())
+	vmi = libwait.WaitUntilVMIReady(vmi, console.LoginToAlpine)
+	// configure the client VMI
+	Expect(libnet.AddIPAddress(vmi, ifaceName, ipCIDR)).To(Succeed())
+	return vmi
+}
+
+func createAlpineVMIRandomNode(networkName string, mac string) (*v1.VirtualMachineInstance, error) {
```

Comment:
All those helper method have very similar code. I would add a basic builder method for the common called, and each method will call the builder with the relevant params.

e.g - ` createVMI(nodeName *string, networkName string, ifaceName string, mac *string, os string)`